sycl/source/detail/pi_opencl.cpp

```diff
@@ -78,6 +78,57 @@ pi_result OCL(piextDeviceSelectBinary)(
   return PI_SUCCESS;
 }
 
+pi_result OCL(piQueueCreate)(pi_context context, pi_device device,
+                             pi_queue_properties properties, pi_queue *queue) {
+  PI_ASSERT(queue, "piQueueCreate failed, queue argument is null");
+
+  cl_platform_id curPlatform;
+  cl_int ret_err = clGetDeviceInfo(pi_cast<cl_device_id>(device),
+                                   CL_DEVICE_PLATFORM, sizeof(cl_platform_id),
+                                   &curPlatform, NULL);
+
+  if (ret_err != CL_SUCCESS) {
+    *queue = nullptr;
+    return pi_cast<pi_result>(ret_err);
```

Comment:
I noticed that the _pi_result enumerator was a little sparse at the moment and I'm casting the return code for a few of these calls and the error codes aren't always in the enumerator, I suppose at worst its undefined behavior that's unlikely to break anything in this case. 

However, would you like me to populate the _pi_result enumerator with PI equivalents for the invocations I am making in this case?

Reply:
I think this is not so important now, most of the calls just bail out on every error now.
But if you are still making changes then that would be a good clean up.